### PR TITLE
Refactor old code and make Claim Types pretty

### PR DESCRIPTION
**Pull request by rasmusguterstam**
# Jira Issue: []

## What & why
- A lot of codescouting
- Refactors old code on Claim Type
- Just a lot of minor bugs and fixes
- A prettier design
- Once we're going for sub types and flags, there's no need for heavy refactoring - we can just extend on this
- With that said, this PR has nothing directly to do with the Claim Types project, and can therefore be deployed independently

### New
https://user-images.githubusercontent.com/51323202/121544343-ee82d480-ca09-11eb-9230-e8b850d05b62.mov

Screenshot 2021-06-10 at 16 18 51

### Old

image

Screenshot 2021-06-10 at 16 15 27

## Optional checklist
- [x] Updated `src/changelog.ts`
- [x] Codescouted
- [ ] Unit tests written
- [x] Tested locally

### CODE REVIEW

File: src/components/claims/claim-details/components/ClaimType.tsx [hunk 1]
```diff
@@ -68,241 +71,247 @@ const hasTicket = (typename: ClaimTypes): boolean => {
...
+      )}
+      {hasPoliceReport(type.__typename as ClaimTypes) && (
+        <DataField style={{ marginTop: '1.0em' }}>
+          <Label>Police report</Label>
+          <Input
+            value={formData.policeReport}
+            onChange={(e) =>
+              setFormData({ ...formData, policeReport: e.target.value })
+            }
+            placeholder={'Has a police report been filed?'}
```
Comment (vonElfvin) [inline src/components/claims/claim-details/components/ClaimType.tsx:186]: If it's just a question we could, I don't say we should, change this to a checkbox. However, I do think the question should be phrased something like "What is the Police Report ID?" or something

File: src/components/claims/claim-details/components/ClaimType.tsx [hunk 2]
```diff
@@ -68,241 +71,247 @@ const hasTicket = (typename: ClaimTypes): boolean => {
...
+          setDate={(newDate) => setFormData({ ...formData, date: newDate })}
+        />
+      </DataField>
+      {hasItem(type.__typename as ClaimTypes) && (
+        <DataField style={{ marginTop: '1.0em' }}>
+          <Label>Item</Label>
+          <Input
+            value={formData.item}
+            onChange={(e) => setFormData({ ...formData, item: e.target.value })}
+            placeholder={'Any specific item?'}
```
Comment (vonElfvin) [inline src/components/claims/claim-details/components/ClaimType.tsx:174]: Itemizer 😢 
```suggestion
            placeholder={'Any specific item(s)?'}
```

File: src/components/claims/claim-details/components/ClaimType.tsx [hunk 3]
```diff
@@ -68,241 +71,247 @@ const hasTicket = (typename: ClaimTypes): boolean => {
...
+          }).then(() => {
+            setSavedFormData(formData)
+          })
+        }}
+      >
+        {loading
+          ? 'Saving...'
+          : formData !== savedFormData
+          ? 'Save changes'
+          : 'Changes saved'}
```
Comment (vonElfvin) [inline src/components/claims/claim-details/components/ClaimType.tsx:236]: Noice

File: src/components/claims/claim-details/components/ClaimType.tsx [hunk 4]
```diff
@@ -68,241 +71,247 @@ const hasTicket = (typename: ClaimTypes): boolean => {
...
-          onSubmit={async (values) => {
-            if (!values.selectedType) {
-              return
-            }
-            setSetClaimTypeLoading(true)
-            await setClaimType({
-              variables: { id: claimId, type: values.selectedType },
+  return (
+    <Paper>
+      <PaperTitle title={'Claim Type'} badge={titleBadge()} />
```
Comment (vonElfvin) [inline src/components/claims/claim-details/components/ClaimType.tsx:283]: Neat

File: src/components/claims/claim-details/components/ClaimType.tsx [hunk 5]
```diff
@@ -68,241 +71,247 @@ const hasTicket = (typename: ClaimTypes): boolean => {
...
-                  claimInformation: {
-                    ...values,
-                    date: values.date && format(values.date, 'yyyy-MM-dd'),
-                  },
-                },
+            .catch(() => {
+              showNotification({
+                header: 'Error',
+                type: 'red',
+                message: 'Something went wrong, notify the tech team',
```
Comment (vonElfvin) [inline src/components/claims/claim-details/components/ClaimType.tsx:305]: Maybe we can print the message instead
```suggestion
                message: e.message,
```

File: src/components/claims/claim-details/components/ClaimType.tsx [hunk 6]
```diff
@@ -68,241 +71,247 @@ const hasTicket = (typename: ClaimTypes): boolean => {
...
-        ) : null}
-      </Loadable>
+            })
+        }}
+        noOptionsMessage={() => 'No types found'}
+        options={Object.keys(ClaimTypes).map((claimType) =>
+          createClaimTypeOption(claimType),
+        )}
+      />
+      {!!type && <ClaimTypeDataForm type={type} claimId={claimId} />}
```
Comment (vonElfvin) [inline src/components/claims/claim-details/components/ClaimType.tsx:314]: Not this?
```suggestion
      {type && <ClaimTypeDataForm type={type} claimId={claimId} />}
```

File: src/components/claims/claim-details/components/claim-items/PaperTitle.tsx [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+const PaperTitleWrapper = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  margin-bottom: 1em;
+`
+
+export const PaperTitle: React.FC<{
```
Comment (vonElfvin) [inline src/components/claims/claim-details/components/claim-items/PaperTitle.tsx:19]: Not in `hedvig-ui`?

Comment (vonElfvin) [reply src/components/claims/claim-details/components/claim-items/PaperTitle.tsx:19]: Very pretty! Nice design 😍

**Review by vonElfvin — COMMENTED** [review]
Very nice components and design updates! 👍 👍  Would love to see them moved to `hedvig-ui` 😍

File: src/components/claims/claim-details/components/ClaimFileTable.tsx [hunk 8]
```diff
@@ -89,94 +102,93 @@ const ClaimFileTableComponent: React.FC<WithShowNotification & {
...
-          <Table.Row>
-            <Table.HeaderCell>Claim Files</Table.HeaderCell>
-            <Table.HeaderCell>File Type</Table.HeaderCell>
-            <Table.HeaderCell>Uploaded At</Table.HeaderCell>
-            <Table.HeaderCell />
-          </Table.Row>
-        </Table.Header>
-        <Table.Body>
-          {loading && (
+      {claimFiles.length !== 0 && (
```
[the remark this answers is not in the document]
Comment (rasmusguterstam) [reply src/components/claims/claim-details/components/ClaimFileTable.tsx:105]: On line 81 we have

```typescript
const claimFiles = claimFilesData?.claim?.claimFiles ?? []
```

File: src/components/claims/claim-details/components/ClaimType.tsx (hunk 2 above)
Comment (rasmusguterstam) [reply src/components/claims/claim-details/components/ClaimType.tsx:174]: Ikr, breaks my heart 😥

**Pull request merged**